### ISSUE TITLE
Support mongoDb $lookup for populate on assosiated models (Feature Request)

### ISSUE THREAD

**Issue by elennaro** [balderdashy/sails-mongo#375]
Mongo db **[$lookup (aggregation)](https://docs.mongodb.org/manual/reference/operator/aggregation/lookup/)** Performs a left outer join to an unsharded collection in the same database to filter in documents from the “joined” collection for processing.

If the database version is **3.2**, then some simple queries on associated collections could be made with $lookup aggregation.

It also it could be great to add a setting in config/mongo.js(config/orm.js) to make users select preferable way to do population: with or without $lookup (in case they have sharding).

**Comment by particlebanana** [comment]
@elennaro this looks very interesting! It could speed up hasMany queries a lot I'm sure.

Since exposing a new configuration option is a change in behavior/usage rather than a bug fix, this would need to be submitted as a proposal. Would you please review our updated [contribution guide](https://github.com/balderdashy/sails/blob/master/CONTRIBUTING.md#v-proposing-features-and-enhancements), then submit a proposal explaining your use case and how this approach would solve it, as well as much of a spec as you have time for? Beyond the other reasons discussed in our new contribution guide, one of the most important things about understanding your use case is so that we can ensure this is appropriately documented and tested now, and maintained properly in the future.

I appreciate your help and understanding re: the recent changes to our process, and I look forward to learning more about your goals for this enhancement.

**Comment by elennaro** [comment]
Thanks for reply @particlebanana.
I've read the [contribution guide](https://github.com/balderdashy/sails/blob/master/CONTRIBUTING.md#v-proposing-features-and-enhancements) and will try to submit a proposal.

I just have one question.

The CG says that: 

>If it doesn't already exist, create a pull request editing ROADMAP.MD (the easiest way to do this is opening ROADMAP.md while logged in to GitHub and clicking the "Edit" button)... Add a new row to the Backlog table... etc.

Does it mean I should made proposal for **sails-mongo** into **sail's** *ROADMAP.MD*? I didn't find any *ROADMAP.MD* in **sails-mongo**.

P.S. I appreciate you are explaining all the contribution rules to me because It's my first time I do this, and I don't even imagine how to write specs for **sails-mongo**, but I gonna see some examples and will try to do my best.

**Comment by particlebanana** [comment]
@elennaro you can submit it to the [Waterline Roadmap](https://github.com/balderdashy/waterline/blob/master/ROADMAP.md). You don't have to explain all the nitty gritty details but just explain why you need it and when it should be used.

It will use [Waterline-Cursor](https://github.com/balderdashy/waterline-cursor) to do the query. You can see an example in the [Sails-Postgresql](https://github.com/balderdashy/sails-postgresql/blob/master/lib/adapter.js#L615-L891) adapter.

**Comment by elennaro** [comment]
@particlebanana 
I've tried to write a PR: [#1288](https://github.com/balderdashy/waterline/pull/1288).

**Comment by sailsbot** [comment]
Thanks for posting, @elennaro.  I'm a repo bot-- nice to meet you!

It has been 30 days since there have been any updates or new comments on this page.  If this issue has been resolved, feel free to disregard the rest of this message.  On the other hand, if you are still waiting on a patch, please:

  + review our [contribution guide](https://github.com/balderdashy/sails/blob/master/CONTRIBUTING.md) to make sure this submission meets our criteria (only _verified bugs_ with documented features, please;  no questions, commentary, or bug reports about undocumented features or unofficial plugins)
  + create a new issue with the latest information, including updated version details with error messages, failing tests, and a link back to [the original issue]().  This allows GitHub to automatically create a back-reference for future visitors arriving from search engines.

Thanks so much for your help!
